napari/utils/colormaps/colormap.py

```diff
@@ -522,3 +799,47 @@ def minimum_dtype_for_labels(num_colors: int) -> np.dtype:
     if num_colors <= np.iinfo(np.uint16).max:
         return np.dtype(np.uint16)
     return np.dtype(np.float32)
+
+
+try:
+    import numba
+except ModuleNotFoundError:
+    _zero_preserving_modulo = _zero_preserving_modulo_numpy
+    _cast_labels_data_to_texture_dtype_direct_impl = (
+        _cast_labels_data_to_texture_dtype_direct_numpy
+    )
+    prange = range
+else:
+    _zero_preserving_modulo = numba.njit(parallel=True)(
+        _zero_preserving_modulo_jit
+    )
+    _cast_labels_data_to_texture_dtype_direct_impl = (
+        _generate_hash_map_and_cast_labels_data_to_texture_dtype_direct
+    )
+    _cast_direct_labels_to_minimum_type_jit = numba.njit(parallel=True)(
+        _cast_direct_labels_to_minimum_type_jit
+    )
+    prange = numba.prange  # type: ignore [misc]
+
+    del numba
+
+
+@lru_cache(maxsize=128)
+def _primes(upto):
+    """Generate primes up to a given number.
```

Comment:
I think this is the [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) algorithm for generating primes. Regardless, this docstring should explicitly state whatever the algorithm is with a link to a stable reference (Wikipedia is good enough for me).

Reply:
Function was removed